### ISSUE TITLE
Internal error after login

### ISSUE THREAD

**Issue by attie** [attie/redmine_auth_gitlab#2]
Once GitLab refers the user back to Redmine, the following error will be displayed:

![image](https://user-images.githubusercontent.com/2195519/62837690-254efe80-bc6a-11e9-8828-16c142c08a3d.png)

Current URL is: `http://redmine.example.com/oauth2callback?code=XXX`

**Comment by attie** [comment]
```
I, [2019-08-11T18:16:50.720198 #22]  INFO -- : Started GET "/oauth_gitlab?back_url=http%3A%2F%2Fperdy.attie.co.uk%3A3001%2Fprojects" for 10.42.0.18 at 2019-08-11 18:16:50 +0000
I, [2019-08-11T18:16:50.721855 #22]  INFO -- : Processing by GitlabAuthController#oauth_gitlab as HTML
I, [2019-08-11T18:16:50.721969 #22]  INFO -- :   Parameters: {"back_url"=>"http://perdy.attie.co.uk:3001/projects"}
I, [2019-08-11T18:16:50.723930 #22]  INFO -- :   Current user: anonymous
I, [2019-08-11T18:16:50.724393 #22]  INFO -- : Redirected to https://gitlab.agsys.io/oauth/authorize?client_id=47a19ba169bf1d843a8e5ca1eaa3ece3211b0ccde303018226f8c9b8fadaab9a&redirect_uri=http%3A%2F%2Fperdy.attie.co.uk%3A3001%2Foauth2callback&response_type=code&scope=openid
I, [2019-08-11T18:16:50.724452 #22]  INFO -- : Completed 302 Found in 2ms (ActiveRecord: 0.5ms)
10.42.0.18 - - [11/Aug/2019:18:16:50 UTC] "GET /oauth_gitlab?back_url=http%3A%2F%2Fperdy.attie.co.uk%3A3001%2Fprojects HTTP/1.1" 302 292
http://perdy.attie.co.uk:3001/login?back_url=http%3A%2F%2Fperdy.attie.co.uk%3A3001%2Fprojects -> /oauth_gitlab?back_url=http%3A%2F%2Fperdy.attie.co.uk%3A3001%2Fprojects
I, [2019-08-11T18:16:50.780929 #22]  INFO -- : Started GET "/oauth2callback?code=c764eaf7c000c82e266b7cc83a73012f7f6e75500c4bebedfef048a99f6f7237" for 10.42.0.18 at 2019-08-11 18:16:50 +0000
I, [2019-08-11T18:16:50.782082 #22]  INFO -- : Processing by GitlabAuthController#oauth_gitlab_callback as HTML
I, [2019-08-11T18:16:50.782170 #22]  INFO -- :   Parameters: {"code"=>"c764eaf7c000c82e266b7cc83a73012f7f6e75500c4bebedfef048a99f6f7237"}
I, [2019-08-11T18:16:50.788126 #22]  INFO -- :   Current user: anonymous
I, [2019-08-11T18:16:50.932457 #22]  INFO -- : Completed 500 Internal Server Error in 150ms (ActiveRecord: 2.0ms)
F, [2019-08-11T18:16:50.933962 #22] FATAL -- :
F, [2019-08-11T18:16:50.934046 #22] FATAL -- : OAuth2::Error (insufficient_scope: The request requires higher privileges than provided by the access token.
{"error":"insufficient_scope","error_description":"The request requires higher privileges than provided by the access token.","scope":"read_user api"}):
F, [2019-08-11T18:16:50.934117 #22] FATAL -- :
F, [2019-08-11T18:16:50.934169 #22] FATAL -- : plugins/redmine_auth_gitlab/app/controllers/gitlab_auth_controller.rb:23:in `oauth_gitlab_callback'
lib/redmine/sudo_mode.rb:63:in `sudo_mode'
10.42.0.18 - - [11/Aug/2019:18:16:50 UTC] "GET /oauth2callback?code=c764eaf7c000c82e266b7cc83a73012f7f6e75500c4bebedfef048a99f6f7237 HTTP/1.1" 500 648
http://perdy.attie.co.uk:3001/ -> /oauth2callback?code=c764eaf7c000c82e266b7cc83a73012f7f6e75500c4bebedfef048a99f6f7237
```

**Comment by attie** [comment]
```
I, [2019-08-11T18:16:50.720198 #22]  INFO -- : Started GET "/oauth_gitlab?back_url=http%3A%2F%2Fredmine.example.com%2Fprojects" for 10.42.0.18 at 2019-08-11 18:16:50 +0000
I, [2019-08-11T18:16:50.721855 #22]  INFO -- : Processing by GitlabAuthController#oauth_gitlab as HTML
I, [2019-08-11T18:16:50.721969 #22]  INFO -- :   Parameters: {"back_url"=>"http://redmine.example.com/projects"}
I, [2019-08-11T18:16:50.723930 #22]  INFO -- :   Current user: anonymous
I, [2019-08-11T18:16:50.724393 #22]  INFO -- : Redirected to https://gitlab.agsys.io/oauth/authorize?client_id=XXX&redirect_uri=http%3A%2F%2Fredmine.example.com%2Foauth2callback&response_type=code&scope=openid
I, [2019-08-11T18:16:50.724452 #22]  INFO -- : Completed 302 Found in 2ms (ActiveRecord: 0.5ms)
10.42.0.18 - - [11/Aug/2019:18:16:50 UTC] "GET /oauth_gitlab?back_url=http%3A%2F%2Fredmine.example.com%2Fprojects HTTP/1.1" 302 292
http://redmine.example.com/login?back_url=http%3A%2F%2Fredmine.example.com%2Fprojects -> /oauth_gitlab?back_url=http%3A%2F%2Fredmine.example.com%2Fprojects
I, [2019-08-11T18:16:50.780929 #22]  INFO -- : Started GET "/oauth2callback?code=YYY" for 10.42.0.18 at 2019-08-11 18:16:50 +0000
I, [2019-08-11T18:16:50.782082 #22]  INFO -- : Processing by GitlabAuthController#oauth_gitlab_callback as HTML
I, [2019-08-11T18:16:50.782170 #22]  INFO -- :   Parameters: {"code"=>"YYY"}
I, [2019-08-11T18:16:50.788126 #22]  INFO -- :   Current user: anonymous
I, [2019-08-11T18:16:50.932457 #22]  INFO -- : Completed 500 Internal Server Error in 150ms (ActiveRecord: 2.0ms)
F, [2019-08-11T18:16:50.933962 #22] FATAL -- :
F, [2019-08-11T18:16:50.934046 #22] FATAL -- : OAuth2::Error (insufficient_scope: The request requires higher privileges than provided by the access token.
{"error":"insufficient_scope","error_description":"The request requires higher privileges than provided by the access token.","scope":"read_user api"}):
F, [2019-08-11T18:16:50.934117 #22] FATAL -- :
F, [2019-08-11T18:16:50.934169 #22] FATAL -- : plugins/redmine_auth_gitlab/app/controllers/gitlab_auth_controller.rb:23:in `oauth_gitlab_callback'
lib/redmine/sudo_mode.rb:63:in `sudo_mode'
10.42.0.18 - - [11/Aug/2019:18:16:50 UTC] "GET /oauth2callback?code=YYY HTTP/1.1" 500 648
http://redmine.example.com/ -> /oauth2callback?code=YYY
```

**Comment by attie** [comment]
```
I, [2019-08-11T18:16:50.720198 #22]  INFO -- : Started GET "/oauth_gitlab?back_url=http%3A%2F%2Fredmine.example.com%2Fprojects" for 10.42.0.18 at 2019-08-11 18:16:50 +0000
I, [2019-08-11T18:16:50.721855 #22]  INFO -- : Processing by GitlabAuthController#oauth_gitlab as HTML
I, [2019-08-11T18:16:50.721969 #22]  INFO -- :   Parameters: {"back_url"=>"http://redmine.example.com/projects"}
I, [2019-08-11T18:16:50.723930 #22]  INFO -- :   Current user: anonymous
I, [2019-08-11T18:16:50.724393 #22]  INFO -- : Redirected to https://gitlab.example.com/oauth/authorize?client_id=XXX&redirect_uri=http%3A%2F%2Fredmine.example.com%2Foauth2callback&response_type=code&scope=openid
I, [2019-08-11T18:16:50.724452 #22]  INFO -- : Completed 302 Found in 2ms (ActiveRecord: 0.5ms)
10.42.0.18 - - [11/Aug/2019:18:16:50 UTC] "GET /oauth_gitlab?back_url=http%3A%2F%2Fredmine.example.com%2Fprojects HTTP/1.1" 302 292
http://redmine.example.com/login?back_url=http%3A%2F%2Fredmine.example.com%2Fprojects -> /oauth_gitlab?back_url=http%3A%2F%2Fredmine.example.com%2Fprojects
I, [2019-08-11T18:16:50.780929 #22]  INFO -- : Started GET "/oauth2callback?code=YYY" for 10.42.0.18 at 2019-08-11 18:16:50 +0000
I, [2019-08-11T18:16:50.782082 #22]  INFO -- : Processing by GitlabAuthController#oauth_gitlab_callback as HTML
I, [2019-08-11T18:16:50.782170 #22]  INFO -- :   Parameters: {"code"=>"YYY"}
I, [2019-08-11T18:16:50.788126 #22]  INFO -- :   Current user: anonymous
I, [2019-08-11T18:16:50.932457 #22]  INFO -- : Completed 500 Internal Server Error in 150ms (ActiveRecord: 2.0ms)
F, [2019-08-11T18:16:50.933962 #22] FATAL -- :
F, [2019-08-11T18:16:50.934046 #22] FATAL -- : OAuth2::Error (insufficient_scope: The request requires higher privileges than provided by the access token.
{"error":"insufficient_scope","error_description":"The request requires higher privileges than provided by the access token.","scope":"read_user api"}):
F, [2019-08-11T18:16:50.934117 #22] FATAL -- :
F, [2019-08-11T18:16:50.934169 #22] FATAL -- : plugins/redmine_auth_gitlab/app/controllers/gitlab_auth_controller.rb:23:in `oauth_gitlab_callback'
lib/redmine/sudo_mode.rb:63:in `sudo_mode'
10.42.0.18 - - [11/Aug/2019:18:16:50 UTC] "GET /oauth2callback?code=YYY HTTP/1.1" 500 648
http://redmine.example.com/ -> /oauth2callback?code=YYY
```

**Comment by attie** [comment]
Turns out we needed the `read_user` scope too... fixed in c2eaa92
